### PR TITLE
1757 - Datagrid Clear/Delete Rows

### PR DESCRIPTION
**Pull request by tjamesallen15**
**Explain the _details_ for making this change. What existing problem does the pull request solve?**

This pull request will fix clear button not deleting or clearing rows from other pages.

**Related github/jira issue (required)**:
Closes https://github.com/infor-design/enterprise-wc/issues/1757

**Steps necessary to review your pull request (required)**:
- Pull this git repository, build, and run the app
- Go to http://localhost:4300/ids-data-grid/pagination-client-side.html
- Select one row
- Go to other page
- Try any combination of page number, page size, and selection.
- Click delete button
- Do it again with clear button

**Included in this Pull Request**:
- [ ] An e2e or functional test for the bug or feature.
- [x] A note to the change log.

### CODE REVIEW

File: src/core/ids-data-source.ts [hunk 1]
```diff
@@ -411,6 +411,20 @@ class IdsDataSource {
     this.#total = this.#currentData.length;
   }
 
+  /**
+   * Query row data from the dataset
+   * @param {Array<Record<string, unknown>>} row records to get
+   * @returns {object} The dataset object
+   */
+  query(row: number) {
```
Comment (tmcconechy) [inline src/core/ids-data-source.ts:419]:
```suggestion
  query(row: number) {
```

Can we call this `findById`? It is finding by ID right?

**Review by tmcconechy — CHANGES_REQUESTED** [review]
@tjamesallen15 I think the changes to updateDataset could cause issues. Im a little concerned about including a change in that method to fix this until we get the tests in place.

1. Can you look for a fix that is just in the example or in removeRow?
2. Additionally i see some cases not working
a) select 4 then delete , then select 5 and delete (this doesnt work)
b) select 4 then got to last page then delete, then go to first and the select 5 and then go to last page then delete (this doesnt work)

May want to punt on this issue as its not critical and need to get tests back in place but give it a try this way?

**Comment by tjamesallen15** [comment]
> @tjamesallen15 I think the changes to updateDataset could cause issues. Im a little concerned about including a change in that method to fix this until we get the tests in place.
> 
> 1. Can you look for a fix that is just in the example or in removeRow?
> 2. Additionally i see some cases not working
>    a) select 4 then delete , then select 5 and delete (this doesnt work)
>    b) select 4 then got to last page then delete, then go to first and the select 5 and then go to last page then delete (this doesnt work)
> 
> May want to punt on this issue as its not critical and need to get tests back in place but give it a try this way?

Yeah. My bad. It's my first ticket in WC. I'll try to rethink the solution on this one.

**Comment by tmcconechy** [comment]
@tjamesallen15 no problem, take your time. I am also working on converting tests so that will help.

**Review by tmcconechy — DISMISSED** [review]
Looks good @tjamesallen15 can you add a test for this in tests/ids-data-grid?

Thanks!!!

**Comment by tjamesallen15** [comment]
> Looks good @tjamesallen15 can you add a test for this in tests/ids-data-grid?
> 
> Thanks!!!

Glad you like it! Will do!

**Comment by tmcconechy** [comment]
@tjamesallen15 if you pull main it should fix the test for you

**Review by tmcconechy — APPROVED** [review]

**Review by jdilag90 — APPROVED** [review]

**Pull request merged**